backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CAdapterGenerator.kt

```diff
@@ -639,9 +638,21 @@ internal class CAdapterGenerator(
         return true
     }
 
+
+    private val moduleDescriptors = mutableSetOf<ModuleDescriptor>()
+
     fun generateBindings() {
         scopes.push(ExportedElementScope(ScopeKind.TOP, "kotlin"))
-        context.moduleDescriptor.accept(this, null)
+        moduleDescriptors += context.moduleDescriptor
+        moduleDescriptors += context.getExportedDependencies()
+
+        currentPackageFragments = context.moduleDescriptor.getPackageFragments().sortedWith(
```

Comment:
Maybe include exported dependencies' package fragments too?

Reply:
But why, looks like everything is exported anyway (as descriptor tree contains all them).

Reply:
What is `currentPackageFragments` for then?

Reply:
OK, reworked.